### PR TITLE
fix network bridge modules and sysctl

### PR DESCRIPTION
**Pull request by rdxmb**
fix https://github.com/geerlingguy/ansible-role-kubernetes/issues/92

### CODE REVIEW

**Comment by rdxmb** [comment]
if this looks good for you I can create another PR with sqashed commits.

**Comment by rdxmb** [comment]
This could also be part of the containerd-role, like it is described here: https://kubernetes.io/docs/setup/production-environment/container-runtimes/#containerd .

What do you think?

**Comment by gleichda** [comment]
@rdxmb @geerlingguy what about this PR tested it on Ubuntu 21.04 Server on a raspberry Pi and can confirm it works

File: tasks/sysctl-setup.yml [hunk 1]
```diff
@@ -8,14 +8,40 @@
     or ansible_distribution_major_version | int < 10
 
 # See: https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/install-kubeadm/#letting-iptables-see-bridged-traffic
-- name: Let iptables see bridged traffic.
-  sysctl:
```
Comment (geerlingguy) [inline tasks/sysctl-setup.yml:12]: Is there some reason the `sysctl` module can't be used for these modifications (and `lineinfile` used instead)?

Comment (rdxmb) [reply tasks/sysctl-setup.yml:12]: I do not remember.

I've just complied with https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/install-kubeadm/#letting-iptables-see-bridged-traffic .

**Comment by terryzwt** [comment]
I apply the patch mannually, works for me.

**Comment by gleichda** [comment]
@rdxmb see the original issue or https://stackoverflow.com/a/63692277 where @geerlingguy answered the question already on stackoverflow. Validated today with Ubuntu LTS 20.04 on a RPI 4

So for me this could be closed. Not sure if would be a good idea to include the br_netfliter module into this playbook?

**Comment by rdxmb** [comment]
If this works, this sounds good.
However, I prefer using a config-file under `/etc/sysctl.d` , which can also be done via ansible's `sysctl`.

https://docs.ansible.com/ansible/latest/collections/ansible/posix/sysctl_module.html#parameter-sysctl_file

This way a later change / cleanup can be done through ansible by deleting this file.

**Comment by gleichda** [comment]
Sounds valid no objections there. Will you rewrite your PR or should I create a new one!

**Comment by rdxmb** [comment]
feel free to create a new one. Rewrite is bad because of missing squash. Thanks.

File: tasks/sysctl-setup.yml (hunk 1 above)
Comment (aanoaa) [reply tasks/sysctl-setup.yml:12]: This is immediately applied, but will only last 'till the next boot, so it is not a permanent change.
to make sysctl changes permanently write `token = value` formatted file into below dirs.

```
FILES
       /run/sysctl.d/*.conf
       /etc/sysctl.d/*.conf
       /usr/local/lib/sysctl.d/*.conf
       /usr/lib/sysctl.d/*.conf
       /lib/sysctl.d/*.conf
       /etc/sysctl.conf
              The paths where sysctl preload files usually exist.  See also sysctl option --system.
```

**Comment by rdxmb** [comment]
no-stale

@gleichda will you continue with your pr?

**Comment by gleichda** [comment]
@rdxmb unfortunately not at the moment due to missing time. But as soon as I find some spare time I will.
In the meantime if you want to and have time pls go ahead

**Pull request closed without merging**